CenterForOpenScience/osf.io
# [Feature] Add str value conversion to encrypt and decrypt [OSF-7200]

## Purpose

Names on Dropbox and Google drive break when we try to encrypt them.

## Changes

1. Add helper function to ensure incoming strings to be encoded are in the proper utf-8 format
2. Add tests for helper function and encryption in general since it was untested.

## Side effects

None known


## Ticket
https://openscience.atlassian.net/browse/OSF-7200

[#OSF-7200]

## QA Considerations
1. Test auth on at least 1 addon. I did dropbox locally and it worked. Per MattF if it works for one, there is no reason it shouldn't work for all of them. Worth testing at least one or two for a sanity check, but probably not worth checking all of them (especially if you can't easily change your name on the service).
2. Change your name on <addon> to contain unicode madness and try to authorize the addon on the osf
3. Ensure it also works with non-unicode madness
